onnx/defs/tensor/defs.cc

```diff
@@ -958,6 +958,11 @@ ONNX_OPERATOR_SET_SCHEMA(
 
           if (!startsInitializer || !endsInitializer || (hasInputShape(ctx, 3) && !ctx.getInputData(3)) ||
               (hasInputShape(ctx, 4) && !ctx.getInputData(4))) {
+            const auto input_rank = ctx.getInputType(0)->tensor_type().shape().dim_size();
```

Comment:
This looks good, thanks! Unrelated to this change, I wonder if we should change the `hasInputShape` in the previous two lines to `hasInput`? We need to only check if the starts/ends inputs are supplied, not whether we know their shape.

Reply:
Failed. Reverting